src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/Map16FW.java

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright 2016-2020 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.maven.plugin.internal.generated;
+
+import java.nio.ByteOrder;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.reaktor.internal.test.types.Flyweight;
+
+public final class Map16FW<K extends Flyweight, V extends Flyweight> extends MapFW<K, V>
+{
+    private static final int LENGTH_SIZE = BitUtil.SIZE_OF_SHORT;
+
+    private static final int FIELD_COUNT_SIZE = BitUtil.SIZE_OF_SHORT;
+
+    private static final int LENGTH_OFFSET = 0;
+
+    private static final int FIELD_COUNT_OFFSET = LENGTH_OFFSET + LENGTH_SIZE;
+
+    private static final int FIELDS_OFFSET = FIELD_COUNT_OFFSET + FIELD_COUNT_SIZE;
+
+    private static final int LENGTH_MAX_VALUE = 0xFFFF;
+
+    private final ByteOrder byteOrder;
+
+    private final K keyRO;
+
+    private final V valueRO;
+
+    private final DirectBuffer entriesRO = new UnsafeBuffer(0L, 0);
+
+    public Map16FW(
+        K keyRO,
+        V valueRO)
+    {
+        this.keyRO = keyRO;
+        this.valueRO = valueRO;
+        this.byteOrder = ByteOrder.nativeOrder();
+    }
+
+    public Map16FW(
+        K keyRO,
+        V valueRO,
+        ByteOrder byteOrder)
+    {
+        this.keyRO = keyRO;
+        this.valueRO = valueRO;
+        this.byteOrder = byteOrder;
+    }
+
+    @Override
+    public int length()
+    {
+        return buffer().getShort(offset() + LENGTH_OFFSET, byteOrder);
+    }
+
+    @Override
+    public int fieldCount()
+    {
+        return buffer().getShort(offset() + FIELD_COUNT_OFFSET, byteOrder);
+    }
+
+    @Override
+    public void forEach(
+        BiConsumer<K, V> consumer)
+    {
+        int offset = offset() + FIELDS_OFFSET;
+        int fieldCount = fieldCount();
+        for (int i = 0; i < fieldCount; i += 2)
+        {
+            K key = (K) keyRO.tryWrap(buffer(), offset, limit());
+            V value = key != null ? (V) valueRO.wrap(buffer(), key.limit(), limit()) : null;
+            consumer.accept(key, value);
+            if (key == null || value == null)
+            {
+                break;
+            }
```

Comment:
This part should still use `wrap` and not defend against `null` as by this point the `wrap` has been successful, so no need for `tryWrap`.